iris-mpc-common/src/helpers/sync.rs

```diff
@@ -807,13 +838,156 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_check_genesis_config_all_equal() {
+        let genesis_config = Some(GenesisConfig {
+            max_indexation_height: 100,
+            last_indexation_height: 50,
+        });
+
+        let states = vec![
+            SyncState {
+                db_len: 10,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config.clone(),
+            },
+            SyncState {
+                db_len: 20,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config.clone(),
+            },
+            SyncState {
+                db_len: 30,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config.clone(),
+            },
+        ];
+
+        let sync_result = SyncResult::new(states[0].clone(), states);
+        assert!(sync_result.check_genesis_config().is_ok());
+    }
+
+    #[test]
+    fn test_check_genesis_config_not_equal() {
+        let genesis_config_1 = Some(GenesisConfig {
+            max_indexation_height: 100,
+            last_indexation_height: 50,
+        });
+        let genesis_config_2 = Some(GenesisConfig {
+            max_indexation_height: 200,
+            last_indexation_height: 150,
+        });
+
+        let states = vec![
+            SyncState {
+                db_len: 10,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config_1.clone(),
+            },
+            SyncState {
+                db_len: 20,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config_2.clone(),
+            },
+            SyncState {
+                db_len: 30,
+                deleted_request_ids: vec![],
+                modifications: vec![],
+                next_sns_sequence_num: None,
+                common_config: CommonConfig::default(),
+                genesis_config: genesis_config_1.clone(),
+            },
+        ];
+
+        let sync_result = SyncResult::new(states[0].clone(), states);
+        assert!(sync_result.check_genesis_config().is_err());
+    }
+
+    #[test]
+    fn test_check_genesis_config_all_none() {
```

Comment:
A but misleading name here, `all_none` but only one of them is `None`. Is this intentional? seems swapped with the next test case